### PR TITLE
Refactor storage code

### PR DESCRIPTION
**Pull request by volnistii11**

### CODE REVIEW

File: internal/app/server/server.go [hunk 1]
```diff
@@ -4,14 +4,20 @@ import (
...
-func RunServer() {
+type Server struct {
+}
+
+func RunServer(repository storage.Repository) {
 	config.ParseFlags()
 
+	h := handlers.NewHandlerProvider(repository)
+
 	r := gin.Default()
```
Comment (arttet) [inline internal/app/server/server.go:18]:
```
func Router() http.Handler {
	r := chi.NewRouter()

	r.Post("/", shortener)
	r.Get("/{id}", expander)

	return r
}
```

File: internal/app/storage/storage.go [hunk 2]
```diff
@@ -5,37 +5,31 @@ import (
 	"github.com/volnistii11/URL-shortener/internal/app/utils"
 )
 
-type URL struct {
-	URLDependency map[string]string
+type Repository interface {
+	ReadURL(id string) (string, error)
+	WriteURL(url string) string
```
Comment (arttet) [inline internal/app/storage/storage.go:10]: It would be better if you return an error, either.

File: internal/app/storage/storage.go [hunk 3]
```diff
@@ -5,37 +5,31 @@ import (
...
-func (storage *URL) ReadURL(id string) (string, error) {
-	if fullURL, ok := storage.URLDependency[id]; !ok {
+type url struct {
+	urlDependency map[string]string
+}
+
+func (storage *url) ReadURL(id string) (string, error) {
+	if fullURL, ok := storage.urlDependency[id]; !ok {
 		return fullURL, errors.New("full url not found")
 	} else {
```
Comment (arttet) [inline internal/app/storage/storage.go:26]: You don't need to use `else`

**Review by arttet — APPROVED** [review]
Well done!

File: internal/app/server/server.go (hunk 1 above)
Comment (volnistii11) [reply internal/app/server/server.go]: > ```
> func Router() http.Handler {
> 	r := chi.NewRouter()
> 
> 	r.Post("/", shortener)
> 	r.Get("/{id}", expander)
> 
> 	return r
> }
> ```

Hi, what do you mean here?)

**Pull request merged**